bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1070,12 +1070,19 @@ impl<'a> CircuitInputStateRef<'a> {
             return Ok(Some(ExecError::InvalidOpcode));
         }
 
+        let call = self.call()?;
         // When last step has opcodes that halt, there's no error.
         if matches!(next_step, None)
-            && matches!(
+            && (matches!(
```

Comment:
suggest below change for better readiblity 🚀 
```rust
if matches!(next_step, None) { // enumerating root call scope successful cases
       // case 1: root call with normal halt opcode termination
           if matches!(step.op, OpcodeId::STOP | OpcodeId::REVERT | OpcodeId::SELFDESTRUCT) {
               return Ok(None);
           }
       // case 2: root call is NOT Create (Create represented by empty tx.to) and halt by opcode::Return 
          if !call.is_create() && step.op == OpcodeId::RETURN {
              return Ok(None);
          }
       // more other case...
}
```

Side question: is `call.is_root == true` equivalent to `matches!(next_step, None)` ? if yes then also success then I vote `call.is_root == true` for better  readibility

Reply:
good comment， for `call.is_root == true` equivalent to `matches!(next_step, None)`, I think it is not aways, even though for internal call(`call.is_root == false`) next_step is not none in most cases.   there is no ensure for that.

Reply:
add comment in [eaae0e0](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1275/commits/eaae0e01abde0b696fa97313def1bb14f0ef8b75) , removed root call from comment.

Reply:
Thanks for the clarification, I learned it still possible that got edge cases for internal call with `next_step = None`